docs/guidance_for_upgraders.md

```diff
@@ -0,0 +1,74 @@
+# Guidance for Operator Developer
+
+After you change either the operator signature in a BC breaking way or the semantics of the operator, you will need to write an “upgrader” to make the change non-BC breaking iff they are used in TorchScript or mobile. In general, you can know your operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `
+
+The steps to write upgrader:
+
+1. Prepare a model before making changes to the operator, following the process:
+    1. Add a module in `test/jit/fixtures_srcs/fixtures_src.py`. The reason why we generate the model file before updating the operator is that, an old model with the old operator before the change is needed to ensure the upgrader is working as expected. In `test/jit/fixtures_srcs/generate_models.py`, add the module and its corresponding changed operator name like following:
+
+    ```
+    ALL_MODULES = {
+        TestVersionedDivTensorExampleV7(): "aten::div.Tensor",
```

Comment:
Do we have tests to verify that the key to this dictionary is a TestCase instance not a TestCase class? I think one of the common mistake is forgetting those parenthese.

Reply:
Oh very good point! Let me have a pr to add test as well as error message

Reply:
Address in https://github.com/pytorch/pytorch/pull/72615